pdns/ws-api.cc

```diff
@@ -114,22 +114,50 @@ void apiDiscovery(HttpRequest* req, HttpResponse* resp) {
   };
   Json doc = Json::array { version1 };
 
-  resp->setBody(doc);
+  resp->setJsonBody(doc);
+}
+
+/* Return OpenAPI document describing the supported API.
+ */
+void apiDocs(HttpRequest* req, HttpResponse* resp) {
+  if(req->method != "GET")
+    throw HttpMethodNotAllowedException();
+
+  string fname = ::arg()["api-document-file"];
+  if (fname.empty())
+    throw HttpNotFoundException();
+
+#ifdef HAVE_CXX11
+  std::ifstream ifs(fname, std::ifstream::binary);
+#else
+  std::ifstream ifs(fname.c_str(), std::ifstream::binary);
+#endif
```

Comment:
```suggestion
   std::ifstream ifs(fname, std::ifstream::binary);
```

pdns already requires C++11